### PR TITLE
GH-681 Add option to kick a player for being AFK

### PR DESCRIPTION
**Pull request by failutee**
## Description

The introduced change allows for kicking a player from the server due to being AFK.

Fixes # (issue)
#681

### CODE REVIEW

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/afk/AfkController.java [hunk 1]
```diff
@@ -14,10 +26,39 @@
...
+
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    void onAfkSwitch(AfkSwitchEvent event) {
+        UUID playerUUID = event.getAfk().getPlayer();
+
+        if (!this.afkService.isAfk(playerUUID) || !this.config.afk.kickOnAfk) {
+            return;
+        }
+
+        Player player = Bukkit.getPlayer(playerUUID);
```
Comment (vLuckyyy) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/afk/AfkController.java:51]: Use Server.

**Review by vLuckyyy — APPROVED** [review]

**Review by Rollczi — CHANGES_REQUESTED** [review]

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/afk/AfkController.java [hunk 2]
```diff
@@ -14,10 +26,39 @@
...
 
     private final AfkService afkService;
+    private final MiniMessage miniMessage;
+    private final UserManager userManager;
+    private final TranslationManager translationManager;
+    private final PluginConfiguration config;
 
     @Inject
-    AfkController(AfkService afkService) {
+    AfkController(AfkService afkService, MiniMessage miniMessage, UserManager userManager, TranslationManager translationManager, PluginConfiguration config) {
```
Comment (vLuckyyy) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/afk/AfkController.java:35]:
```suggestion
    AfkController(
    AfkService afkService,
    MiniMessage miniMessage,
    UserManager userManager,
    TranslationManager translationManager, 
    PluginConfiguration config
   ) {
```

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/afk/AfkController.java
Comment (vLuckyyy) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/afk/AfkController.java:1]: Can you make this in two classes? AfkStateController and AfkKickController?

Comment (failutee) [reply eternalcore-core/src/main/java/com/eternalcode/core/feature/afk/AfkController.java:1]: Sure

**Review by P1otrulla — APPROVED** [review]
Do what they tell you and it will be great

**Comment by failutee** [comment]
Have I done something wrong? Because for me, it builds normally without any problems
@vLuckyyy

**Comment by vLuckyyy** [comment]
> Have I done something wrong? Because for me, it builds normally without any problems @vLuckyyy

Fixed

**Review by Rollczi — APPROVED** [review]

**Pull request merged**